app/src/main/java/org/jellyfin/androidtv/ui/playback/PlaybackController.java

```diff
@@ -122,6 +122,14 @@ public PlaybackController(List<BaseItemDto> items, IPlaybackOverlayFragment frag
         useVlc = userPreferences.getValue().get(UserPreferences.Companion.getVideoPlayer()) == PreferredVideoPlayer.VLC;
     }
 
+    public void clearFragment() {
+        mFragment = null;
```

Comment:
Is this meant to "destroy" the fragment? Because if that's the case you need to destroy it via the fragmentmanager, otherwise the fragment will still be bound to the activity and cause memory issues.

Reply:
Kind of. I mean it was supposed to but it didn't so I added the `hasFragment` function so we only run the `onDestroy` code if we haven't cleared the fragment. The fragment is still getting destroyed though beacause it's onDestroy function is still getting called. It's just getting called after we want it to. I was going to look for a better way to do all this but then figured we will rewrite it eventually AND this guarantees things are done when we want them to be.